etcd/requests.go

```diff
@@ -303,8 +303,8 @@ func DefaultCheckRetry(cluster *Cluster, numReqs int, lastResp http.Response,
 	err error) error {
 
 	if numReqs > 2*len(cluster.Machines) {
-		return newError(ErrCodeEtcdNotReachable,
-			"Tried to connect to each peer twice and failed", 0)
+		errStr := fmt.Sprintf("Tried to connect to members %v twice and failed: %v", cluster.Machines, err)
```

Comment:
@thockin The function is used to check whether retry is reasonable. It can be called in two cases: 

1. there is an connection error. When there is a connection error, it deserves to do requests on other etcd members and it may succeed.

2. it receives a http response with bad status, e.g., StausInternalServerError, which may indicate the cluster is under leader election and we should wait for a while before retry.

It is reasonable to retry in both cases.

the `connect to` word is inaccurate, I will improve it.